### PR TITLE
[Feature] Implement undo controller

### PR DESCRIPTION
**Pull request by alexandraFlavia9**
This PR adds undo/redo controller methods

## Related tickets

-   [WRS-630](https://support.chili-publish.com/projects/WRS/issues/WRS-630)

### CODE REVIEW

File: src/controllers/SubscriberController.ts [hunk 1]
```diff
@@ -97,4 +97,14 @@ export class SubscriberController {
         const callBack = this.config.onSelectedToolChanged;
         callBack(tool as ToolType);
     };
+
+    /**
+     * Listener on state changes
+     * @param undoStackState Stringified object of UndoStackState
+     */
+    onUndoStackStateChanged = (undoStackState: string) => {
+        console.log('here');
```
Comment (alitunaarel) [inline src/controllers/SubscriberController.ts:106]: Console.log can be removed?

File: src/controllers/UndoController.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { EditorAPI } from '../../types/CommonTypes';
+
+/**
+ * The UndoController is responsible for all communication regarding the Undo-Manager.
+ * Methods inside this controller can be called by `window.SDK.undoManager.{method-name}`
+ */
+export class UndoController {
```
Comment (rizikolik) [inline src/controllers/UndoController.ts:7]: I think we should ask this controller Name to the Product.It includes undo and redo but we only call it undoController.

File: src/controllers/SubscriberController.ts [hunk 3]
```diff
@@ -97,4 +97,13 @@ export class SubscriberController {
         const callBack = this.config.onSelectedToolChanged;
         callBack(tool as ToolType);
     };
+
+    /**
+     * Listener on state changes
+     * @param undoStackState Stringified object of UndoStackState
+     */
+    onUndoStackStateChanged = (undoStackState: string) => {
```
Comment (rizikolik) [inline src/controllers/SubscriberController.ts:105]: Names should align with Product I think.For me its fine.

File: src/controllers/UndoController.ts (hunk 2 above)
Comment (brapoprod) [reply src/controllers/UndoController.ts:7]: Kind of true, also on editor-engine it's called UndoService, so controller can have the same name (UndoServiceController).
On this.x level it should indeed be validated by product. SDK.undo.undo would be a bit weird.

File: src/index.ts [hunk 4]
```diff
@@ -60,6 +61,8 @@ export class SDK {
     tool: ToolController;
     page: PageController;
     debug: DebugController;
+    undoManager: UndoController;
```
Comment (brapoprod) [inline src/index.ts:64]: this is the one that definitely should be validated by product

**Review by rizikolik — APPROVED** [review]

File: src/controllers/SubscriberController.ts [hunk 5]
```diff
@@ -97,4 +97,13 @@ export class SubscriberController {
         const callBack = this.config.onSelectedToolChanged;
         callBack(tool as ToolType);
     };
+
+    /**
+     * Listener on state changes
+     * @param undoState Stringified object of UndoState
+     */
+    onUndoStateChanged = (undoState: string) => {
```
Comment (brapoprod) [inline src/controllers/SubscriberController.ts:105]:
```suggestion
    onUndoManagerChanged = (undoState: string) => {
```

**Review by brapoprod — APPROVED** [review]
minor suggestion

**Pull request merged**